TopPlaces/TopPlaces/Controllers/PlacesTableViewController.m

```diff
@@ -0,0 +1,187 @@
+  for (id<PlaceInfo> place in places) {
+    
+    //add the place under the corresponding country key
+    
+    if (![countryToPlacesMap objectForKey:place.country]) {
+      countryToPlacesMap[place.country] = [[NSMutableArray alloc] init];
+    }
+    
+    [countryToPlacesMap[place.country] addObject:place];
+  }
+  
+  // sort places for each country
+  for (NSString *country in countryToPlacesMap.allKeys) {
+    NSArray *places = countryToPlacesMap[country];
+    NSArray *sortedPlaces = [places sortedArrayUsingComparator:^NSComparisonResult(id a, id b) {
+      id<PlaceInfo> aPlace = (id<PlaceInfo>)a;
+      id<PlaceInfo> bPlace = (id<PlaceInfo>)b;
+      
+      return [aPlace compare:bPlace];
+    }];
+    
+    countryToPlacesMap[country] = sortedPlaces;
+  }
+
+  return countryToPlacesMap;
+}
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(__nullable id)sender {
+  if ([segue.identifier isEqualToString:@"showPhotosOfPlace"]) {
+    NSLog(@"prepareForSegue: showPhotosOfPlaces");
+    
+    TopPhotosTableViewController *tpvc =
+        (TopPhotosTableViewController *)segue.destinationViewController;
+    
+    //get the selected cell row/section
+    NSIndexPath *path = [self.tableView indexPathForCell:(UITableViewCell *)sender];
+    
+    //set the placeInfo for the incoming VC
+    NSString *country = self.sortedContries[path.section];
+    tpvc.placeInfo = self.countryToPlacesMap[country][path.row];
+    
+    //propagate the placePhotosProvider
+    tpvc.placesPhotosProvider = self.placesPhotosProvider;
+  }
+}
+
+#pragma mark -
+#pragma mark table view controller overrides
+#pragma mark -
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+  return [self.sortedContries count];
+}
+
+- (nullable NSString *)tableView:(UITableView *)tableView
+         titleForHeaderInSection:(NSInteger)section {
+  return self.sortedContries[section];
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  NSString *country = self.sortedContries[section];
+  return [self.countryToPlacesMap[country] count];
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+  UITableViewCell *cell;
+  cell = [self.tableView dequeueReusableCellWithIdentifier:@"FlickrPlacesCell"];
+  
+  NSString *country = self.sortedContries[indexPath.section];
+  id<PlaceInfo> placeInfo = self.countryToPlacesMap[country][indexPath.row];
+  cell.textLabel.text = placeInfo.title;
+  cell.detailTextLabel.text = placeInfo.details;
+  return cell;
+}
+
+#pragma mark -
+#pragma mark split view delegates
+#pragma mark -
+
+- (BOOL)splitViewController:(UISplitViewController *)splitViewController
+    collapseSecondaryViewController:(UIViewController *)secondaryViewController
+    ontoPrimaryViewController:(UIViewController *)primaryViewController {
+  return YES; //show master view by default
```

Comment:
Move this comment to be right above the method, and please give few more words why it is needed.